---
title: "Disposition"
author: "Lia Baumann"
date: "2024-04-17"
output: word_document
bibliography: references_bib.json
editor_options: 
  markdown: 
    wrap: 72
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## Topic

"Genetic structure and long-term genetic diversity assessment of
Burgundy truffle populations in central Europe"

## Research question(s) and hypotheses

Prelim.: What is the population structure of the truffles? Are MLGs
perennial? Differences between or within populations? Interactions...
etc?

## Research aims

The Burgundy truffle is an abundant ectomycorrhizal symbiont that occurs
in a wide range of temperate climates. Despite its prized underground
ascocarps, its complex lifecycle and possible response to anthropogenic
climate change are largely known. A citizen science monitoring of
Burgundy truffle ascocarp production was run with a 3-week resolution in
23 natural populations at the center of its European distribution over
up to 11 years. More than 3000 truffles were genotyped using
microsatellite markers to assess the genetic structure and diversity of
these populations in space and time. The aims of this research project
are to assess the population structure, compare intra- and inter
populations and assess the gene flow and perenniality of individuals.

## A review of the current state of research

### Truffles in general

True truffles are ectomycorrhizal fungi belonging to the genus Tuber in
Ascomycetes. This genus, which exclusively forms hypogeous fruiting
bodies (i.e. ascocarps or sporocarps), known as truffles, comprises more
than 200 species all over the world ([@bonito2010], [@bonito2013],
[@guevara2013], [@reyna2014], [@berch2016]), of which at least 30
naturally occur in Europe [@ceruti]. Some of these species are well
known for their organoleptic properties and their high economic value, due to their use in haute cuisine,
such as *Tuber magnatum* (Alba white truffle), *Tuber melanosporum*
(Périgord black truffle), and *Tuber aestivum* Vittad. (Burgundy and
summer truffles, depending on the harvesting period)
[@boaWildEdibleFungi2004]; [@bonito2010]; [@chevaliera], [@reyna2014a].

As ectomycorrhizal fungi, truffles form symbiotic associations with
roots of a wide range of ectomycorrhizal tree species [@benucci2011],
[@stobbe2012]. They grant their hosts access to water and nutrients,
protect roots from disease and pest, and interact with and sometimes even
kill understory, non-host plants [@gryndler2014]; [@streiblová2012];
[@taschen2020]).

In contrast to other high-value forest mushrooms, such as *Boletus* spp.,
*Cantharellus* spp. and *Laccaria* spp., which can develop their fruitbodies
during a few days [@moore2008]; [@halbwachs2016]), hypogenous truffle
fruitbodies are expected to grow more slowly [@pacioni2015]. However, it
is still unknown how much time truffles need to develop and how many
generations of fruitbodies can occur within one ‘growing’ season between
the formation of primordia and the realisation of a fully mature and
ripe gleba [@büntgen2017]. The main reasons for the still little understood abiotic and biotic drivers of
truffle growth are the truffle's subterranean life cycle dynamics, complex host interactions and also the lack
of non-destructive investigation techniques [@pacioni2015b].
Finding truffles in situ is not possible without the help of well-trained sniffing dogs
or domestic trufflepigs having the olfactory sense of detecting the volatile aroma compounds
of truffle fruitbodies from a distance of up to 50m [@splivallo2016]. This also means
that our knowledge about the truffle's secret and hidden way of life is restricted to
only the fruitbodies that were managed to be detected [@trappe2010].

"In fact, we simply do not know how selective a harvest is and how well
it represents the actual abundance of fruitbodies since the absence of
evidence is no evidence of absence." @buntgenEcoarchaeologicalExcavationTechniques2021


### *Tuber aestivum*

*T. aestivum* is indigenous to many European countries and has a widespread distribution range from Sweden to
Spain, and has also been found outside Europe in North Africa [@stobbe2013], [@jeandroz2008]. Since the 1970s, *T. aestivum* as well as
some other truffle species have been successfully cultivated in truffle
orchards @chevalierCultivationBurgundyTruffle.

Most past studies have focused on the species' aroma profiles (@splivalloSmellTrufflesAroma2016,
@TruffleVolatilesChemical), taxonomic status (@molinierFirstIdentificationPolymorphic2013), @leonardiTypificationFourMost2021@bonitoImprovedResolutionMajor2010, @bonitoHistoricalBiogeographyDiversification2013) 
and genetic diversity (@molinierSSRbasedIdentificationGenetic2016, @molinierFinescaleSpatialGenetic2015, @staubliHiddenFairyRings2022).

The genome sequencing of *T. melanosporum* in 2010 (@martinPerigordBlackTruffle2010), as well as
numerous other studies of *T. melanosporum* (@TuberMelanosporumMating), *T. magnatum*
(Paolocci et al. 2006;Riccioni et al. 2008;Rubinietal. 2005, 2011a,b)
and *T. indicum* (Belfiori et al. 2013) have revealed that these species
are heterothallic. This has also been confirmed for *T. aestivum* (@payenChapterSevenTruffle2014).
Homothallic species contain both mating types (MAT) in their genome, which makes them self-fertile,
while heterothallic species present only one of the two types in the MAT locus (@billiardSexOutcrossingMating2012).
To achieve its life cycle, a female type, i.e. ascogonium, has to be fertilized by a strain of the opposite
mating type. In truffles, a fruiting body is composed of peridium, gleba
and ascospores. The gleba is a haploid tissue consisting of the maternal
parent genotype, which is usually also found as mycelia in the
surrounding soil and on nearby ECM root tips (@muratFinescaleSpatialGenetic2013; @TuberMelanosporumMating, Rubini
et al. 2011a). Ascospores, on the other hand, originate from the recombination
of the maternal and paternal genotypes [@letacon2015].

The popularity of truffles as an exclusive ingredient of elite
gastronomy has risen tremendously in recent years (Hall et al., 2003).
Not only the highly prized Alba white truffle(Tuber magnatum), and the
Périgord black truffle(Tuber melanosporum), but also the more common
Burgundy truffle(Tuber aestivum) are coveted. In contrast to T. magnatum
and T. melanosporum, which are native to Mediterranean environments
(Chevalier and Sourzat, 2012; Vasquez et al., 2014), T. aestivum is
distributed across large parts of temperate Europe (Chevalier, 2010;
Stobbe et al., 2012).

According to Hall et al. (2003, 2007), *T. aestivum* is the most common edible fungus species
in Europe. --> check that again!


Based on inventories of hypogeous fungi in the region of Basel, Switzerland, it has even been
estimated to be among the most abundant mushroom species in mixed beech
forests on calcareous soils (Schw€arzel, 1967). Nevertheless, T.
aestivum has only been sparsely harvested in many countries, and is even
legally protected from harvest in Germany for its alleged rarity (Stobbe
et al., 2013a). This discrepancy in the estimation of its abundance is
due to the fragmentary knowledge about the species' occurrence and
distribution, which, in turn, is related to its hidden life belowground
and the difficulty ofdetecting its hypogeous fruit bodies. The great
demand for T. aestivum has increased cultivation attempts of this
symbiotic ascomycete fungus (Murat, 2015). Although truffle plantations
are now established over much of the species' natural distribution range
(Stobbe et al., 2013a), setting up new plantations is costly and
investments have to be made on a long-term horizon since growth of
truffle mycelium in soil is slow (Molinier et al., 2016) and harvests
can begin no earlier than 5 y after the initial inoculation of host
seedlings, reaching their maximum about a decade after planting (Callot,
1999). Like many other truffle species, T. aestivum grows predominantly
in soils with a high content of limestone or exchangeable calcium
(Chevalier and Sourzat, 2012) and forms ectomycorrhizas with a wide
range of tree and shrub species (Stobbe et al., 2012, 2013a).

@moserEcologicalIndicatorsTuber2017

### Results from WSL

Previous studies utilizing random-amplified polymorphic DNA (RAPD) or
intersimple sequence repeat (ISSR) markers and sequencing of a few genes
suggested the existence of genetic diversity between *T. aestivum*
populations (Gandeboeuf et al., 1997 ; Mello et al., 2002 ; Weden et
al., 2004 ). Direct shotgun pyrosequencing (DSP) has been successfully
used for the identification of microsatellites in animals or plants but,
to our knowledge, this approach has never been used for ectomycorrhizal
fungi. Recently the sequencing of the black truffle genome revealed a
substantial richness in repeated sequences such as microsatellites (
Murat et al., 2011 ). This result suggested that a DSP approach could be
used for identifying microsatellites in Tuber species. The aim of our
study was therefore to use DSP to identify, for the first time,
polymorphic microsatellites in the economically important truffle *T.
aestivum* (@molinierFirstIdentificationPolymorphic2013).

Among Tuber species, *Tuber aestivum* Vittad. has a wide distributional
range being found naturally all over Europe. A study by
@molinierSSRbasedIdentificationGenetic2016 performed large-scale
population genetic analyses in *T. aestivum* to (i) investigate its
genetic diversity at the European scale, (ii) characterize its genetic
structure and test for the presence of ecotypes and (iii) shed light
into its demographic history. To reach these goals, 230 ascocarps from
different populations were genotyped using 15 polymorphic simple
sequence repeat markers. They identified 181 multilocus genotypes and
four genetic groups which did not show a clear geographical separation;
although, one of them was present exclusively in Southeast France, Italy
and Spain. Fixation index values between pairs ofgenetic groups were
generally high and ranged from 0.29 to 0.45. A significant deficit of
heterozygosity indicated a population expansion instead of a recent
population bottleneck, suggesting that *T. aestivum* is not endangered
in Europe, not even in Mediterranean regions. This study based on a
large-scale population genetic analysis suggests that genetically
distinct populations and likely ecotypes within *T. aestivum* are
present.

@buntgenEcoarchaeologicalExcavationTechniques2021 introduced a novel
eco-archaeological approach to study in situ aspects of subterranean
life cycle dynamics and genetic structures of the Burgundy truffle in
three different natural habitats in Baden-Württemberg, southern Germany.
They applied a suite of fine-scale excavation techniques to reveal
high-resolution, spatiotemporal snapshots of the composition, size and
developmental stage of a wide-range of individual truffles. Our approach
not only exposes fruitbodies in the investigated soil layers, but also
relates each of them to its edaphic environment, and further describes
the obtained genotype composition.

@steidingerFallSummerTruffle2022 presented 8 years (2011–2018) of
citizen science monitoring data on the fruitbody production of naturally
occurring T. aestivum populations. Using machine learning, they modeled
how truffle productivity is influenced by soil physicochemical, fungal
meta-community, climate, and host tree phenology. Additionally, they
characterized the broad species climatic envelope of European summer
truffles. By comparing the sensitivity of summer truffle fruitbody
productivity to interannual climate trends, they tested whether these
center-of-range, keystone populations are locally sensitized to climate
change.

@molinierFinescaleGeneticStructure2016 determined mating type and simple
sequence repeat (SSR) maternal genotypes of mapped fruiting bodies to
assess their genetic structure within two naturally colonized forest
sites in southern Germany. Forty-one genotypes were identified from 112
fruiting bodies. According to their mating types, the maternal genotypes
were aggregated only in one population. Genotypic diversity of
individuals that mostly were small and occurred in 1 out of 2 years of
sampling was high. Although these results suggested a ruderal
colonization strategy, some genets spread several hundred meters. This
result indicates that, besides sexual spore dispersal, vegetative growth
or spreading by mycelial propagules contributes to dissemination. In one
site, fewer individuals with a tendency to expand genets belonging to
only one genetic group were observed. In the second site, numerous small
individuals were found and were grouped into two clearly differentiated
genetic groups that were spatially intermingled. Forest characteristics
and disturbances are possible reasons for the observed genetic patterns.

@moserEcologicalIndicatorsTuber2017 aimed at estimating the potential of
using vegetation composition as an additional parameter to detect and
describe *T. aestivum* habitats, and subsequently use this information
to define suitable cultivation sites. We, therefore, assess the
abundance of individual plant species at 16 sites with known T. aestivum
occurrences in southern Germany and Switzerland. We compare the
floristic composition of these sites with that of a systematic sample of
232 Swiss F. sylvatica, Carpinus betulus and Ostrya carpinifolia (FCO)
forest sites with a climate that is, according to current knowledge,
favourable for T. aestivum.

## Identification of gaps in the field of research, motivation for the research and justification of the added scientific value for the field of study

Much remains unknown about the life history of these iconic species due
to a lack of available data from natural truffle populations.

However, many aspects of the subterranean life cycle of the genus, such
as the formation, duration, maturation and deterioration of their
fruitbodies, remain enigmatic and require innovative scientific
endeavours (Büntgen and Egli, 2014; Büntgen et al., 2017).
Interdisciplinary and international research projects are now indicating
that Burgundy truffles (Tuber aestivum) are more widely distributed in
ecologically suitable habitats across central Europe than previously
recognised (Stobbe et al., 2012; Molinier et al., 2016c; Cejka et al.,
2020; Puliga et al., 2021).

Sequencing of the T. aestivum genome is currently being completed (Payen
et al., 2014), and the first sequences have facilitated the development
of specific simple sequence repeat markers (SSR) (Molinier et al.
2013a). A first assessment of the species’ genetic structure at the
European-scale revealed the existence ofwell-differentiated sympatric
genetic groups, indicating different ecotypes (defined as genetically
distinct varieties within a species that are adapted to specific
environmental conditions) with reduced gene flow, although this previous
study provided no clues about how these ecotypes are spatially
distributed (Molinier et al. 2015b). At much smaller scales, a recent
study that focused on the link between genetic structure and aroma in an
orchard naturally colonized by T. aestivum indicated that genet sizes
(up to 92 m) are larger than those of T. melanosporum (Molinier et al.
2015a;Murat et al. 2013). However, the limited sample size did not allow
any deep genetic structure analyses, and the mating-type genotype
distribution was not considered. @molinierFinescaleGeneticStructure2016

## Materials and Methods

Data collection

We selected a total of 20 natural *T. aestivum* populations from within the center of its known natural geographic distribution to monitor truffle fruitbody production (Figure 2a). These populations were defined as spatially restricted locations of known truffle occurrences ranging in size from 20 to 1000 m2, depending on the contiguous area under compatible host trees where trained truffle dogs found *T. aestivum* fruitbodies. Monitoring occurred every 3 weeks throughout the whole year by citizen scientists using dogs trained to scent summer truffles.. At each site, only one and the same citizen-dog team performed the monitoring throughout the whole project. For all belowground fruitbodies detected by the truffle dogs, the soil was roughly removed and the species identity, mass (g), and degree of maturity were assessed. The degrees of maturity were divided into the following qualitative categories:

1.  unripe: white, hard, no or indistinct odor

2.  semi ripe: light beige-light brown, hard, indistinct (nutty) odor

3.  ripe: brown to dark brown, hard, intense odor

(@steidingerFallSummerTruffle2022)

-   Molecular preparation

Genomic DNA of the fruiting bodies was isolated from gleba of each
sample using DNeasy® Plant Mini Kit (Qiagen, Hilden, Germany) according
to the manufacturer’s instructions. Extracts were eluted in 50 μl
ofBuffer AE supplied as part of the kit. To identify the mating type of
each fruiting body, specific primer pairs were designed for each MAT
idiomorph based on T. aestivum genome sequences (Accession numbers:
LT593973 TuaestMAT1-1 gene (European Nucleotide Archive); JB402662.1
TuaestMAT1-2 gene (Genbank)); aest-MAT1-1f (5′ CTACATTCTGGTGG GCGATT
3′)/aest-MAT1-1r (5′ TCCCGATTTGTCCA ACGTAT 3′) and aest-MAT1-2f (5′
ATCGTCGGGACTCA TCTCAC 3′)/aest-MAT1-2r (5′ CGGATATTGGGATT TGATGG 3′).
For mating-type gene amplification, multiplex PCRs were performed in a
total volume of25 μl consisting of 2.5 μl of 10× REDTaq PCR Buffer (11
mM MgCl2;Sigma: B5926), 5 μl dNTP, (1mM, Life Technologies), 1 μl
multiplex solution containing the four primers (10 μMeach), 1.25 μl RED
Taq Polymerase, (1 U/μl, Sigma: D4309), 13.25 μlof sterile water and 2
μl template DNA diluted 10 times. The PCR reactions were performed in a
Veriti® thermal cycler (Applied Biosystems, Foster City, CA, USA) with
the following conditions: 2 min at 94 °C followed by 28 cycles of
denaturation at 94 °C for 30 s, annealing at 57 °C for 30 s, extension
at 72 °C for 1 min, and a final extension at 72 °C for 7 min. PCR
products were run on 1.5 % agarose gel and visualized with a UV
transilluminator after ethidium bromide staining.

The extracted DNA from fruiting bodies was amplified using a set of 14
SSR loci (aest01, aest06, aest07, aest10, aest15, aest18, aest24,
aest25, aest26, aest28, aest29, aest31, aest35 and aest36) developed
previously by [@molinier2013]. Polymerase chain reactions were carried
out using the Qiagen Multiplex PCR kit (Qiagen, Germany) following the
manufacturer’s instructions. PCRs were performed in a total volume of 7
μl, consisting of 3.5 μlof Qiagen Multiplex Buffer (2×), 1.8 μl of
sterile water, 0.7 μlprimer premix (2 μM) and 1 μl template DNA diluted
10 times. Two different primer premixes were used according to the
expected allelic sizes to avoid overlapping. Both multiplex mixtures had
the same PCR conditions: samples were denatured at 95 °C for 15 min,
followed by 28 cycles consisting ofa denaturing step at 94 °C for 30 s,
an annealing step at 60 °C for 1 min and an extension step at 72 °C for
1 min. A final extension step at 60 °C for 30 min was added after 28
cycles. For the subsequent genotyping step on ABI-3130 (Applied
Biosystems, USA), PCR products were diluted in pure water (1/4) and then
mixed with HiDi Formamide. As an internal size standard, GeneScan™ 500
LIZ™ dye Size Standard (ThermoFisher Scientific, USA) was used.
@molinierFinescaleGeneticStructure2016

-   Genetic analyses

Identification of repeated multilocus genotypes

A multilocus genotype (MLG) was assigned to each fruiting body. To
distinguish true clones from samples carrying the same MLG by chance,
the probability that copies of a MLG arose from sexual reproduction
(Psex) was calculated using MLGSIM (Stenberg et al. 2003). Psex values
were generated by 10,000 simulations using a Monte Carlo simulation
method. A significant Psex value suggests that multiple copies of the
same MLG arose from asexual reproduction and growth (true clones). In
contrast, a non-significant Psex value suggests that the number of
occurrences observed for a repeated MLG resulted from sexual
reproduction (by chance). In the latter case, the occurrences were
treated as different individuals.

The number of MLGs and genotypic diversity, as well as versus the number
of loci, were calculated for one member ofeach MLG using Multilocus 1.3
(Agapow and Burt 2001). Given the clonality observed, subsequent
analyses were conducted on a global data set including all individuals
(ramet level) and on a clone-corrected data set including one member of
each MLG (simple clone-corrected dataset).

For some spatial analyses (i.e. aggregation index, autocorrelation
analyses), we assigned up to four centred location points per MLG for
large MLGs (>15 m) in the clone-corrected data set (partial
clone-corrected dataset) (Fig. S2).
@molinierFinescaleGeneticStructure2016

Persistence of individuals

Comparison of genetic diversity (rarefied)

PCAs

STRUCTURE


## Expected results

Since Tuber spp. are hypogeous, their propagation mainly depends on
insects and larger animals as spore vectors (Trappe and Claridge 2005).
Hypogeous species are thought to have a shorter distance dispersion
potential than epigeous species, which have air-dispersed spores, with
related effects on the gene flow and genetic structure. The consequence
of a short dispersal distance is greater population differentiation and
isolation by distance at smaller spatial scales for hypogeous fungi such
as Rhizopogon occidentalis, R. vulgaris and Tuber melanosporum (Murat et
al. 2013;Grubisha etal. 2007). A fine-scale population structure
analysis carried out in two T. melanosporum plantations using fruiting
bodies and mycorrhizas demonstrated a profound isolation by distance in
the first 5 m and found that belowground genets had a maximum size ofa
fewmeters (Murat et al. 2013). The observed patterns were assumed to be
a combination of vegetative mycelial growth of a few genets that
persisted for several years and an annual recruitment of new genets via
the ascospores. Intriguingly, genotypes were not spatially randomly
distributed in T. melanosporum plantations and natural fields but rather
occurred in clusters according to their MAT idiomorph (Murat et al.
2013;Rubini etal. 2011a). Such a spatial segregation ofmating types
would decrease the probability that compatible cells meet, which could
be beneficial under the assumption that sex is costly (Selosse et al.
2013). This segregation ofthe mating-type strains raises the question
ofwhere paternal individuals come from and what structures they might
consist of (Le Tacon et al., 2015).
@molinierFinescaleGeneticStructure2016

@molinierFinescaleGeneticStructure2016 determined mating type and simple
sequence repeat (SSR) maternal genotypes of mapped fruiting bodies to
assess their genetic structure within two naturally colonized forest
sites in southern Germany. Forty-one genotypes were identified from 112
fruiting bodies. According to their mating types, the maternal genotypes
were aggregated only in one population. Genotypic diversity of
individuals that mostly were small and occurred in 1 out of 2 years of
sampling was high. Although these results suggested a ruderal
colonization strategy, some genets spread several hundred meters. This
result indicates that, besides sexual spore dispersal, vegetative growth
or spreading by mycelial propagules contributes to dissemination. In one
site, fewer individuals with a tendency to expand genets belonging to
only one genetic group were observed. In the second site, numerous small
individuals were found and were grouped into two clearly differentiated
genetic groups that were spatially intermingled. Forest characteristics
and disturbances are possible reasons for the observed genetic patterns.

## Risk assessment

Errors in sampling: risk low.
Errors in sample preparation (PCRs): moderate.
Errors in attribution of markers in GeneMapper: moderate to high.

## Project schedule

-   Data collection almost finished
-   Data analysis: splitting into different parts

## Budget plan

-   Costs: personnel, analyses

## References
